test/viewmodels/DutyDefermentStatementForEoriSpec.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels
+
+import models.DDStatementType.Weekly
+import models.{DutyDefermentStatementFile, DutyDefermentStatementFileMetadata, FileFormat}
+import models.FileRole.DutyDefermentStatement
+import util.SpecBase
+
+import java.time.LocalDate
+
+class DutyDefermentStatementForEoriSpec extends SpecBase {
+
+  "groups" must {
+
+    "correctly give statements by month which are less than 7 months" in new Setup {
+
+      val currentStatementsTestData: Seq[DutyDefermentStatementFile] = List(
+        DutyDefermentStatementFile("someFilename", "downloadUrl", fileSizeData, metaData01),
+        DutyDefermentStatementFile("someFilename2", "downloadUrl", fileSizeData, metaData02),
+        DutyDefermentStatementFile("someFilename3", "downloadUrl", fileSizeData, metaData03)
+      )
+      val requestedStatementsTestData: Seq[DutyDefermentStatementFile] = List(
+        DutyDefermentStatementFile("someFilename", "downloadUrl", fileSizeData, metaData01)
+      )
+
+      lazy val dutyDefermentStatementsForEori: DutyDefermentStatementsForEori = DutyDefermentStatementsForEori(
+        eoriHistory,
+        currentStatementsTestData,
+        requestedStatementsTestData
+      )
+
+      assert(dutyDefermentStatementsForEori.currentStatements.size == 3)
```

Comment:
Please make use of either shouldBe or mustBe instead of using assert. Same for other lines